Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -135,4 +134,107 @@ class CustomerIOImplementation: CustomerIOInstance {
     ) {
         DataPipeline.shared.trackMetric(deliveryID: deliveryID, event: event, deviceToken: deviceToken)
     }
+
+    // TODO: Write test case
+    func handleQueueBacklog() {
+        let allStoredTasks = backgroundQueue.getAllStoredTasks()
+        if allStoredTasks.count <= 0 {
+            logger.info("No tasks pending in the background queue to be executed.")
+            return
+        }
+        threadUtil.runBackground { [weak self] in
+            allStoredTasks.forEach { task in
+                self?.getAndProcessTask(for: task)
+            }
+        }
+    }
+
+    // TODO: Write test case
+    /**
+     Retrieves a task from the queue based on its metadata.
+     Fetches `type` of the task and processes accordingly
+     */
+    func getAndProcessTask(for task: QueueTaskMetadata) {
+        guard let taskDetail = backgroundQueue.getTaskDetail(task) else { return }
+        let taskData = taskDetail.data
+        var isProcessed = true
+        switch taskDetail.taskType {
+        case .trackDeliveryMetric:
+            // TODO: Segment doesn't provide this method by default needs to get added
+            // Remove isProcessed when the method is added
+            print("Track Delivery Metrics for in-app - Needs discussion")
```

Comment:
its the same method as `track metric` for push, we just need to exclude `recipient` 

 ```
let properties: [String: Any] = metaData.mergeWith([
            "metric": event.rawValue,
            "deliveryId": deliveryId,
        ])
```

Reply:
The calls are automatically going to `trackMetric`. What I have been thinking about is the use case of `trackDeliveryMetric` as I could not figure out a way to test this case. I added a "need discussion" comment because this is one of the possible cases in `QueueTaskType` but I had a hard time trying to reproduce this one.